core/commands/storage/session.go

```diff
@@ -41,13 +48,44 @@ type Chunk struct {
 	Receiver  peer.ID
 	Price     int64
 	State     string
+	Proof     string
 	Time      time.Time
 	Err       error
+
+	RetryChan chan *StepRetryChan
+}
+
+type StepRetryChan struct {
+	CurrentStep int
+	Succeed     bool
+	ClientErr   error
+	HostErr     error
 }
 
 func init() {
 	GlobalSession = &SessionMap{}
 	GlobalSession.Map = make(map[string]*Session)
+	StdChunkStateFlow[0] = &ChunkFlowControl{
```

Comment:
You should define 0...6 consts inside this file so we know the step const instead of 0